### PR TITLE
feat(sql): multiple database

### PR DESCRIPTION
**Pull request by xzf0587**
1. add multiple database bicep 
2. update adding database user for multiple database

### CODE REVIEW

**Comment by xzf0587** [comment]
bicep snippet for database module.
image

**Comment by xzf0587** [comment]
bicep snippet for provision
image

File: packages/fx-core/src/plugins/resource/sql/constants.ts [hunk 1]
```diff
@@ -108,4 +109,7 @@ export class AzureSqlBicep {
 export class AzureSqlBicepFile {
   static readonly moduleTemplateFileName: string = "provision.template.bicep";
   static readonly ProvisionModuleTemplateFileName = "sqlProvision.template.bicep";
+  static readonly newDatabaseModuleTemplateFileName: string =
+    "newDatabase.provision.template.bicep";
+  static readonly newDatabaseProvisionModuleTemplateFileName = "newDatabase.template.bicep";
```
Comment (wenytang-ms) [inline packages/fx-core/src/plugins/resource/sql/constants.ts:114]: The two values and names don't seem to match

File: packages/fx-core/src/plugins/resource/sql/plugin.ts [hunk 2]
```diff
@@ -249,9 +250,18 @@ export class SqlPluginImpl {
     managementClient: ManagementClient
   ): Promise<void> {
     let retryCount = 0;
+    const databases: { [key: string]: boolean } = {};
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/sql/plugin.ts:253]: better naming? my understading is this indicates whether add user succeedded for each database?

File: packages/fx-core/src/plugins/resource/sql/plugin.ts [hunk 3]
```diff
@@ -446,4 +493,20 @@ export class SqlPluginImpl {
...
+      if (k.startsWith(Constants.databaseName)) {
+        this.config.databases.push(v);
+      }
+    });
+  }
+
+  private removeDatabases(ctx: PluginContext) {
+    ctx.config.forEach((v: string, k: string) => {
+      if (k.startsWith(Constants.databaseName) && k !== Constants.databaseName) {
+        ctx.config.delete(k);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/sql/plugin.ts:508]: can we merge loadDatabases and removeDatabases into one function?

File: packages/fx-core/src/plugins/resource/sql/constants.ts (hunk 1 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/constants.ts]: Sure. Updated it. Thanks.

File: packages/fx-core/src/plugins/resource/sql/plugin.ts [hunk 4]
```diff
@@ -271,6 +281,9 @@ export class SqlPluginImpl {
   }
 
   public async generateArmTemplates(ctx: PluginContext): Promise<Result<any, FxError>> {
+    if (ctx.answers?.existingResources?.includes(Constants.pluginFullName)) {
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/sql/plugin.ts:284]: minor, can we see which is better?
1. Check existingResources in plugin.ts as what we're doing here
2. Check existingResources in `generateArmTemplates` in index.ts, and call addNewDatabase() from there directly.

File: packages/fx-core/src/plugins/resource/sql/plugin.ts (hunk 3 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/plugin.ts:508]: Actually, loadDatabases is used in both pre-provision and post-provision.
removeDatabases is used in pre-provision stage before loadDatabases. It is used to remove possible outdated database name. It is an additional action.

File: packages/fx-core/src/plugins/resource/sql/plugin.ts (hunk 2 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/plugin.ts]: Thanks.
It is fixed.

File: packages/fx-core/src/plugins/resource/sql/plugin.ts (hunk 4 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/plugin.ts]: Sure. 
Move the check in index.ts.

File: packages/fx-core/src/plugins/resource/sql/plugin.ts [hunk 5]
```diff
@@ -311,6 +324,40 @@ export class SqlPluginImpl {
     return ok(result);
   }
 
+  public async addNewDatabase(ctx: PluginContext): Promise<Result<any, FxError>> {
```
[the remark this answers is not in the document]
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/plugin.ts]: It is fixed.
Thanks.

**Review by blackchoey — APPROVED** [review]

**Pull request merged**